Assets/MixedRealityToolkit/_Core/Utilities/EdgeUtilities.cs

```diff
@@ -0,0 +1,99 @@
+        /// </summary>
+        /// <param name="geometryEdges">The geometry for which we are checking the point.</param>
+        /// <param name="point">The point being checked.</param>
+        /// <returns>
+        /// True if the point falls within the geometry, false otherwise.</returns>
+        public static bool IsInsideBoundary(
+            Edge[] geometryEdges, 
+            Vector2 point)
+        {
+            if (geometryEdges.Length == 0)
+            {
+                return false;
+            }
+
+            // Check if a ray to the right (X+) intersects with an 
+            // odd number of edges (inside) or an even number of edges (outside)
+            var rightEdge = new Edge(point, new Vector2(maxWidth, point.y));
+
+            int intersections = 0;
+            foreach (Edge edge in geometryEdges)
+            {
+                if (IsValidPoint(GetIntersectionPoint(edge, rightEdge)))
+                {
+                    ++intersections;
+                }
+            }
+
+            return (intersections & 1) == 1;
+        }
+
+        /// <summary>
+        /// Checks to see if a point is valid.
+        /// </summary>
+        /// <param name="point">The point to check.</param>
+        /// <returns>True if the point is valid, false otherwise./returns>
+        /// <remarks>
+        /// A point is considered invalid if any one of it's coordinate values are infinite or not a number.
+        /// </remarks>
+        public static bool IsValidPoint(Vector2 point)
+        {
+            return ((!float.IsInfinity(point.x) && !float.IsInfinity(point.y)) &&
+                    (!float.IsNaN(point.x) && !float.IsNaN(point.y)));
+        }
+
+        /// <summary>
+        /// Returns the point at which two <see cref="Edge"/> values intersect.
+        /// </summary>
+        /// <param name="edgeA">The first edge</param>
+        /// <param name="edgeB">The second edge</param>
+        /// <returns>
+        /// A Vector2 representing the point at which the two edges intersect, InscribedRectangleDescription.InvalidPoint otherwise.
+        /// </returns>
+        public static Vector2 GetIntersectionPoint(Edge edgeA, Edge edgeB)
+        {
+            float sA_x = edgeA.PointB.x - edgeA.PointA.x;
+            float sA_y = edgeA.PointB.y - edgeA.PointA.y;
+            float sB_x = edgeB.PointB.x - edgeB.PointA.x;
+            float sB_y = edgeB.PointB.y - edgeB.PointA.y;
+
+            float s = (-sA_y * (edgeA.PointA.x - edgeB.PointA.x) + sA_x * (edgeA.PointA.y - edgeB.PointA.y)) / (-sB_x * sA_y + sA_x * sB_y);
+            float t = (sB_x * (edgeA.PointA.y - edgeB.PointA.y) - sB_y * (edgeA.PointA.x - edgeB.PointA.x)) / (-sB_x * sA_y + sA_x * sB_y);
+
+
+            if ((s >= 0) && (s <= 1) && (t >= 0) && (t <= 1))
+            {
+                // Collision detected
+                return new Vector2(edgeA.PointA.x + (t * sA_x), edgeA.PointA.y + (t * sA_y));
```

Comment:
depending on how often we get the intersection point, we may see a perf increase if we use a cached vector2 to do the calc.

Reply:
this is how the june18_dev branch is coded. i honestly dont know this code well enough to want to change it too much

Reply:
example: 
```
private static Vector2 calcVector = Vector2.zero;

/// <summary>
/// Returns the point at which two <see cref="Edge"/> values intersect.
/// </summary>
/// <param name="edgeA">The first edge</param>
/// <param name="edgeB">The second edge</param>
/// <returns>
/// A Vector2 representing the point at which the two edges intersect, InscribedRectangleDescription.InvalidPoint otherwise.
/// </returns>
public static Vector2 GetIntersectionPoint(Edge edgeA, Edge edgeB)
{
    float sA_x = edgeA.PointB.x - edgeA.PointA.x;
    float sA_y = edgeA.PointB.y - edgeA.PointA.y;
    float sB_x = edgeB.PointB.x - edgeB.PointA.x;
    float sB_y = edgeB.PointB.y - edgeB.PointA.y;

    float s = (-sA_y * (edgeA.PointA.x - edgeB.PointA.x) + sA_x * (edgeA.PointA.y - edgeB.PointA.y)) / (-sB_x * sA_y + sA_x * sB_y);
    float t = (sB_x * (edgeA.PointA.y - edgeB.PointA.y) - sB_y * (edgeA.PointA.x - edgeB.PointA.x)) / (-sB_x * sA_y + sA_x * sB_y);


    if ((s >= 0) && (s <= 1) && (t >= 0) && (t <= 1))
    {
        // Collision detected
        calcVector.x = edgeA.PointA.x + (t * sA_x);
        calcVector.y = edgeA.PointA.y + (t * sA_y);
        return calcVector;
    }

    return InvalidPoint;
}
```